torch/_prims/__init__.py

```diff
@@ -2963,4 +2967,87 @@ def _fft_c2r_aten(
     doc=_fft_c2r_doc,
 )
 
+
+#
+# Statically shaped dynamic operations
+#
+
+
+def _nonzero_static_meta(
+    a: TensorLikeType,
+    *,
+    size: int,
+    fill_value: int = -1,
+) -> TensorLikeType:
+    # Check if size is an non-negative integer
+    utils.check(
+        size >= 0,
+        lambda: f"'size' must be an non-negative integer but got {size}",
+        ValueError,
+    )
+
+    if size == 0:
+        # Return empty tensor
+        return torch.empty(0, dtype=torch.long)
```

Comment:
should `size=0` return 1d or 2d tensor? Regular `nonzero` returns 2d tensor if there are no non-zero elements, with the size (0, ndim)

Reply:
I think the returned tensor size needs to match the specified `size` strictly.

Reply:
`size` is an integer that indicates number of non-zero elements, so returned tensor size (2d) can't match `size`.

Reply:
Yeah, it's an argument of uniformity. If you get (3, ndim) at size=3, (2, ndim) at size=2 and (1, ndim) at size=1, you should get (0, ndim) at size=0.